File: Go/remove_duplicates_from_sorted_arr/26.go
/**
 * Runtime: 6ms (Beats 83.61%)
 * Memory: 4.4MB (Beats 98.13%)
 */

package removeduplicatesfromsortedarr

func removeDuplicates(nums []int) int {
	k := 0
	for i := 1; i < len(nums); i++ {
		if nums[i] != nums[k] {
			k++
			nums[k] = nums[i]
		}
	}
	return k + 1
}

File: TypeScript/rm_dup_from_sorted_arr/26.ts
function removeDuplicates(nums: number[]): number {
  let k = 0;
  for (let i = 1; i < nums.length; i++) {
    if (nums[k] !== nums[i]) {
      k++;
      nums[k] = nums[i];
    }
  }
  return k + 1;
}

export default removeDuplicates;
